overlord/snapstate/autorefresh.go

```diff
@@ -407,6 +413,38 @@ func canContinueAutoRefresh(st *state.State) (int, bool) {
 	return 0, false
 }
 
+func (m *autoRefresh) restoreMonitoring() error {
+	if m.restoredMonitoring {
+		return nil
+	}
+
+	var monitored []string
+	if err := m.state.Get("monitored-snaps", &monitored); err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+
+	defer func() { m.restoredMonitoring = true }()
+	if len(monitored) == 0 {
+		return nil
+	}
+
+	monitoring := make(map[string]chan<- bool)
+	for _, snap := range monitored {
+		done := make(chan string, 1)
+		if err := cgroupMonitorSnapEnded(snap, done); err != nil {
+			return fmt.Errorf("cannot monitor for snap closure: %w", err)
```

Comment:
Is the intention to stop all monitoring if there is cgroup monitoring issue for any one of the snaps?